[Jeffrey Tan] iP

jeffreytjs opened the pull request.

Commit:
Add Gradle support

jeffreytjs committed:
Level-1

Greet, Echo, Exit

jeffreytjs committed:
Level-2

Add, List

jeffreytjs committed:
Level-3

Mark as Done
Extension: A-Classes, use a class to represent tasks

Introduced switch cases and validation of task index for done function

jeffreytjs committed:
Level-4

ToDos, Events, Deadlines

Updated Task to support inheritance of the following three types of tasks:
ToDos: tasks without any date/time attached to it e.g., visit new theme park

Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm

Events: tasks that start at a specific time and ends at a specific time e.g., team project meeting on 2/10/2019 2-4pm

jeffreytjs committed:
Updated runtest script

Set encoding to UTF-8 to resolve symbols appearing as question marks

Corrected pathing for javac

jeffreytjs committed:
A-TextUiTesting

Added sample input commands for testing

Updated corresponding expected output

jeffreytjs committed:
Level-5

Handle Errors
Introduced DukeException class

jeffreytjs committed:
Level-6

Delete
Added "delete" case

Handled a few more exception scenarios

jeffreytjs committed:
Major refactoring of code and improved Exception handling

Abstraction of functions into methods inside helper class
Handled exception more elegantly without terminating the program
Added more test cases (including deleting and exception handling)

jeffreytjs committed:
Implement Save and Load file

Loads tasks from file at start, creates a new
file if one does not yet exist.

jeffreytjs committed:
Implement Level-8 Using dates/times

Update Deadline and Events to take in DateTime
in the following format YYYY/MM/DD HHMM.
Time is optional.

jeffreytjs committed:
Merge pull request #1 from jeffreytjs/branch-Level-7

Merge branch-7

jeffreytjs committed:
Merge branch 'master' into branch-Level-8

jeffreytjs committed:
Merge pull request #2 from jeffreytjs/branch-Level-8

Merge branch-8 into master

siddarth2824 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,83 @@
+                        System.out.println(e.getMessage());
+                    }
+                    break;
+
+                case "todo":
+                case "deadline":
+                case "event":
+                    if (line.length == 1) {
+                        System.out.println("☹ OOPS! Description for command '" + operation + "' not found, try again!");
+                        break;
+                    } else if (operation.equals("todo")) {
+                        try {
+                            Helper.addToDo(line, taskList);
+                        } catch (DukeException e) {
+                            System.out.println(e.getMessage());
+                        }
+                    } else if (operation.equals("deadline")) {
+                        try {
+                            Helper.addDeadline(line, taskList);
+                        } catch (DukeException e) {
+                            System.out.println(e.getMessage());
+                        }
+                    } else {
+                        try {
+                            Helper.addEvent(line, taskList);
+                        } catch (DukeException e) {
+                            System.out.println(e.getMessage());
+                        }
+                    }
+                    break;
+
+                case "delete":
+                    try {
+                        Helper.deleteTask(line, taskList);
+                    } catch (DukeException e) {
+                        System.out.println(e.getMessage());
+                    }
+                    break;
+
+                default:
```
I think according to the Java coding standard guide on the CS2103/T website, under the layout section, it says that there is no indentation for the `case` clauses. Hence I think to avoid any coding standard violation, it would be advisable to fix the indentation for all of the `case` clauses to follow the guide.

jeffreytjs replied:
Thank you for pointing it out, I have realized this common mistake when I did my reviews for others too!

siddarth2824 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,83 @@
+//            System.out.println(t);
+//        }
+        Scanner scanner = new Scanner(System.in);
+        boolean stop = false;
```
According to the Java coding standard guide, under the Naming section, it says that boolean variables/method should sound like booleans. Should this be `boolean isStop` or `boolean shouldStop`  instead of `boolean stop`?

jeffreytjs replied:
This makes sense, thanks!

siddarth2824 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,83 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private static ArrayList<Task> taskList = new ArrayList<>();
```
According to the Java coding standard guide, under the Naming section, they mentioned that naming of constant variables must be all uppercase using underscore to separate words. Hence should this be ` private static ArratList<Task> TASK_LIST` instead of `private static ArrayList<Task> taskList`?

chiamyunqing replied:
Personally, I think naming it as taskList is alright here. I think in the textbook when they are referring to constant variables, it's more like fixed values e.g. int MATH_PI = 3.14 . In this context, I would consider taskList more of a variable rather than a constant instead. :)

jeffreytjs replied:
Yea, thank you both as taskList is meant to be a variable here because it will get updated

siddarth2824 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,83 @@
+            String[] line = scanner.nextLine().split(" ", 2);
+            String operation = line[0];
+            switch (operation) {
+                case "bye":
+                    stop = true;
+                    System.out.println("Bye bye. See you again soon!");
+                    break;
+
```
According to the Java coding standard guide, under the Layout section, I think it shows a sample formatting of a switch statement and in it, there is no blank newline after the `break`. Should there be a blank newline after the `break` in the following code?

jeffreytjs replied:
I'm not too sure about this either, but I should remove it to keep my code clean since it doesn't improve readability also

siddarth2824 commented on src/main/java/Helper.java:
```diff
@@ -0,0 +1,122 @@
+import java.time.format.FormatStyle;
+import java.util.ArrayList;
+
+public class Helper extends Duke{
```
Should there be whitespace in-between `Duke` and `{`?

jeffreytjs replied:
Yes, good spot!

siddarth2824 commented on src/main/java/Helper.java:
```diff
@@ -0,0 +1,122 @@
+        }
+    }
+
+    public static void markAsDone(String[] line, ArrayList<Task> taskList) throws DukeException{
```
Should there be whitespace in-between`DukeException` and `{`?

jeffreytjs replied:
Yes too! Thanks!

siddarth2824 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,83 @@
+
+                default:
+                    // Handles all other inputs
+                    System.out.println("☹ Sorry, I don't recognise that command! Try one of the following instead: todo, event, deadline, done or delete");
```
I think according to the Java coding standard guide, under the Layout section, it says that line length should be no longer than 120 characters. Should the String in the `System.out.println` maybe be wrapped to avoid such a coding standard violation?

jeffreytjs replied:
I've resolved it by setting autowrap so it should not happen anymore!

siddarth2824 commented on src/main/java/ReadWriteFile.java:
```diff
@@ -0,0 +1,73 @@
+
+    public static void readTaskList(ArrayList<Task> taskList) throws DukeException {
+        java.nio.file.Path path = java.nio.file.Paths.get("data").resolve("duke.txt");
+        boolean dirExists = java.nio.file.Files.exists(path);
```
According to the Java coding standard guide, under the Naming section, it says that boolean variables/method should sound like booleans. Should the variable name be maybe renamed to `boolean doesDirExists` or `boolean isDirExists` instead of `boolean dirExists`?

jeffreytjs replied:
You're right thanks!

siddarth2824 reviewed (commented):
Good job on ensuring that there are minimal and not so big Java coding standard violations. One thing which you could maybe think of is to put every class in a package which the Java coding standard guide specifies.

chiamyunqing commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    protected String by;
```
Use nouns for things. Consider more descriptive names such as 'deadlineDate' instead of 'by'.

jeffreytjs replied:
Will do, thanks!

chiamyunqing commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+public class Event extends Task {
+
+    protected String at;
```
Again, use nouns for things. Consider naming this variable as 'eventTime' or something rather than 'at'.

jeffreytjs replied:
Will do as well thanks!

chiamyunqing commented on src/main/java/Helper.java:
```diff
@@ -0,0 +1,122 @@
+        taskList.add(task);
+        ReadWriteFile.writeToFile(taskList);
+
+        System.out.println("Got it. I've added this task:");
```
You may wish to consider combining the 3 "System.out.println(..)" into one e.g. 
```
System.out.println("Got it. I've added this task:" 
                    + task + "Now you have "
                    + taskList.size() + " task(s) in the list.");
```

jeffreytjs replied:
Oh right, thanks for the code example as well. It sure helps! 👍

chiamyunqing commented on src/main/java/ReadWriteFile.java:
```diff
@@ -0,0 +1,73 @@
+                while (sc.hasNext()) {
+                    String[] task = sc.nextLine().split("\\|");
+                    switch (task[0].trim()) {
+                        case "T":
```
Same issue pointed out by the first reviewer. Take note that there's no indentation for your case blocks.

chiamyunqing reviewed (commented):
Overall, code looks neat. However, be careful of the naming of your variables and start doing level 9 onwards! :)

jeffreytjs committed:
Created package Task

jeffreytjs committed:
Bug fix for saving

Correct format for existing tasks.
Fix bug that save done task as 0.
Fix coding standard violations.

jeffreytjs committed:
A-More OOP

Refactor the code to extract out closely related code as classes.

jeffreytjs committed:
Even More OOP and Packages

Fix bug on Event/Deadline parsing and saving

Integrate Helper.java into respective classes

jeffreytjs committed:
Code clean up

jeffreytjs committed:
Add JUnit tests for Parser and ToDo classes

jeffreytjs committed:
Create Jar file

jeffreytjs committed:
Add JavaDoc comments to command package

jeffreytjs committed:
Add JavaDoc comments to task package

Improve method name for saving tasks

jeffreytjs committed:
Add JavaDoc to exception, parser and storage package

jeffreytjs committed:
Add JavaDoc to main duke and ui package

jeffreytjs committed:
Refactor code to align with Coding standard

jeffreytjs committed:
Implement Find feature

Find a task by searching for a keyword

jeffreytjs committed:
Merge branch 'master' into HEAD

jeffreytjs committed:
Merge branch 'temp'

jeffreytjs committed:
Merge branch 'branch-A-CodingStandard'

jeffreytjs committed:
Merge branch 'Level-9'

jeffreytjs committed:
Merge branch 'add-gradle-support' of https://github.com/jeffreytjs/ip

jeffreytjs committed:
Set up Gradle and use CheckStyle

Improved code quality with CheckStyle

Add Javadocs to code in Find feature

jeffreytjs committed:
Implement GUI with JavaFX

jeffreytjs committed:
Modify code for GUI and CheckStyle

jeffreytjs committed:
Modify code for GUI and CheckStyle

jeffreytjs committed:
Merge branch 'Level-10'

jeffreytjs committed:
Bug fix for GUI

Fix GUI freeze when handling user input

jeffreytjs committed:
Add assertions

jeffreytjs committed:
Merge pull request #3 from jeffreytjs/branch-A-Assertions

Add assertions

jeffreytjs committed:
Add extension C-Help

jeffreytjs committed:
Merge pull request #4 from jeffreytjs/branch-C-Help

Add extension C-Help

jeffreytjs committed:
Update Ui for extension C-Help

jeffreytjs committed:
Merge pull request #5 from jeffreytjs/branch-C-Help

Update Ui for extension C-Help

jeffreytjs committed:
Bug fix and update Javadocs

Fix duplicate greeting message and grammar of Javadocs

jeffreytjs committed:
Add product screenshot

jeffreytjs committed:
Add user guide

jeffreytjs committed:
Update README.md

jeffreytjs committed:
Improve GUI

Fix text being truncated.
Update javafx version

jeffreytjs committed:
Merge remote-tracking branch 'origin/master'

jeffreytjs committed:
Handle more exceptions

jeffreytjs committed:
Improve code quality and comply with coding standards

Clean and refactor codes.
Remove unused methods and imports.

jeffreytjs committed:
Acknowledgements

jeffreytjs committed:
Bug fix for GUI

Remove message when file is loaded.
Exit application normally when ByeCommand is give.

jeffreytjs committed:
Rename Ui.PNG to Ui.png for detection

jeffreytjs committed:
Rename Ui.PNG to Ui.png for detection

jeffreytjs committed:
Set theme jekyll-theme-slate

jeffreytjs committed:
Expand valid date formats

jeffreytjs committed:
Merge remote-tracking branch 'origin/master'

jeffreytjs committed:
Remove exit before display bye message

jeffreytjs committed:
Handle exceptions for Done and Delete

Handle when list is empty and when index is more than number of items in list

jeffreytjs committed:
Improve UI messages

jeffreytjs committed:
Update README.md

Add download links to Java JDK 11

jeffreytjs committed:
Update README.md

Set screenshot to centre

jeffreytjs committed:
Update README.md
